### DISCUSSION TITLE
Allow localization of the month + year in the calendar header

### DISCUSSION THREAD

**Discussion by WentTheFox** [mantinedev/mantine#1966] [Q&A] [answered]
According to my translators, in various Asian language the correct order of the calendar header elements would be in a different than the currently hard coded month + year format (year + month is expected). Therefore I'd like to request the ability to configure the date format used in that header, preferably not just the order but via any arbitrary dayjs format sting,

![Date picker component with Japanese locale](https://user-images.githubusercontent.com/81610442/182428252-57a46041-ae22-43ee-a88a-24c87abc51b2.png)

**Comment by rtivital** [comment]
Hi @DJDavid98! you can customize this by providing custom `labelFormat`, for example: `<DatePicker labelFormat="YYYY MMMM" />`. Example on codesandbox – https://codesandbox.io/s/interesting-gould-11ovip?file=/src/App.tsx

**Comment by WentTheFox** [comment]
Hello @rtivital, I was indeed able to match the format requested by my translators by passing `YYYY年MMMM` in case of Japanese, thank you!

However, as far as I can tell, there is no similar property to configure the display of years once that label is clicked. I believe the expectation is that an extra character is added at the end of the year numbers displayed in both the range and on the buttons, which does not seem to be possible currently. I'm not sure if I also missed this and there is a way to do it already, but if not then I'd make this feature request about that instead.

**Reply by rtivital** [reply]
I'll add that with one of the next patches

**Reply by WentTheFox** [reply]
I can see that the `yearLabelFormat` prop is now listed in https://mantine.dev/dates/calendar/?t=props, however, when I set it on the `DatePicker` component in version `5.1.5` it appears to have no effect
![Screenshot of the issue](https://user-images.githubusercontent.com/81610442/184505490-6ae0914a-564a-47c0-8501-8451351d1150.png)

**Reply by rtivital** [reply]
Seems like I've forgotten to add it to DatePicker, and it is exclusive for Calendar component. I'll add it with next patch to DatePicker.

**Reply by WentTheFox** [reply]
While you're there, I just noticed on the screenshot that I actually had localizations for the next/previous a11y labels as well and those do not seem to apply either, I think there may be more properties missing that should be working based on the type definitions.

There's also this message that's being thrown in the console for those labels
![Screenshot of browser console showing the error "Warning: React does not recognize the `nextYearLabel` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `nextyearlabel` instead. If you accidentally passed it from a parent component, remove it from the DOM element."](https://user-images.githubusercontent.com/81610442/184505738-3b757943-1241-458b-92e9-dd0cda763f65.png)

**Comment by WentTheFox** [comment, marked as answer]
Looks like this is fixed with v5.1.6, thank you!
